src/buildExec.ts

```diff
@@ -46,9 +64,20 @@ const buildExec = () => {
   const slug = core.getInput('slug');
   const swift = core.getInput('swift');
   const swiftProject = core.getInput('swift_project');
-  const token = core.getInput('token');
-  const upstream = core.getInput('upstream_proxy');
   const url = core.getInput('url');
+  let token = core.getInput('token');
+  if (token === '') {
+    let codecovURL = url;
+    if (codecovURL === '') {
+      codecovURL = 'https://codecov.io';
+    }
+    try {
+      token = await core.getIDToken(codecovURL);
```

Comment:
@zetaab I think this would do better in a `token.ts` helper file.

More importantly, I think the vast majority of Codecov users aren't using the OIDC and I think this call is unnecessary every time. I would rather have an extra argument added (for now) to explicitly call this function to set the token.

I'd be happy to revisit that in the future if uptick is higher.